### PR TITLE
Add `unimplemented` routine and calls to it in any procedures not yet implemented

### LINKED ISSUE #60: Add procedure `unimplemented` and add calls to it from unimplemented procedures
Add procedure `unimplemented` and add calls to it from unimplemented procedures

### PR DESCRIPTION
**Pull request by ktras**
Closes #60.

### CODE REVIEW

**Review by bonachea — CHANGES_REQUESTED** [review]
Not a blocker for this PR, but tangentially relevant is issue #50 where we'd identify "anything else" that belongs in the implementation with an `unimplemented` tag.

File: src/caffeine/prif_private_s.f90 [hunk 1]
```diff
@@ -7,4 +7,11 @@
   type(prif_team_type), pointer :: current_team => null()
   type(c_ptr) :: non_symmetric_heap_mspace
 
+contains
+
+  subroutine unimplemented(proc_name)
+    character(len=*), intent(in) ::  proc_name
+    error stop proc_name // " is not yet implemented"
```
[the remark this answers is not in the document]
Comment (ktras) [reply src/caffeine/prif_private_s.f90]: Great catch! Updated.

**Comment by ktras** [comment]
> Not a blocker for this PR, but tangentially relevant is issue #50 where we'd identify "anything else" that belongs in the implementation with an `unimplemented` tag.

I have updated issue #50 with clear todo items to be addressed in a future PR.

**Review by bonachea — APPROVED** [review]
LGTM!

**Review by everythingfunctional — APPROVED** [review]

**Pull request merged**